dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.exception.AlreadyExistException;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.workflows.actionlet.NotifyAssigneeActionlet;
+import com.dotmarketing.portlets.workflows.business.DotWorkflowException;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClass;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
+import com.dotmarketing.portlets.workflows.model.WorkflowStep;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.dotmarketing.db.HibernateUtil.addSyncCommitListener;
+
+/**
+ * Helper for Workflow Actions
+ * @author jsanca
+ */
+public class WorkflowHelper {
+
+    private final WorkflowAPI workflowAPI;
+    private final RoleAPI     roleAPI;
+
+    private static class SingletonHolder {
+        private static final WorkflowHelper INSTANCE = new WorkflowHelper();
+    }
+
+    public static WorkflowHelper getInstance() {
+        return WorkflowHelper.SingletonHolder.INSTANCE;
+    }
+
+    private WorkflowHelper() {
+        this( APILocator.getWorkflowAPI(), APILocator.getRoleAPI());
+    }
+
+    @VisibleForTesting
+    protected WorkflowHelper(final WorkflowAPI workflowAPI,
+                             final RoleAPI     roleAPI) {
+
+        this.workflowAPI = workflowAPI;
+        this.roleAPI     = roleAPI;
+    }
+
+
+    /**
+     * Reorder the action associated to the scheme.
+     * @param workflowReorderActionStepForm WorkflowReorderBean
+     */
+    @WrapInTransaction
+    public void reorderAction(final WorkflowReorderBean workflowReorderActionStepForm,
+                              final User user)  {
+
+        WorkflowAction action = null;
+        WorkflowStep step     = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the actionId: "
+                    + workflowReorderActionStepForm.getActionId());
+            action =
+                    this.workflowAPI.findAction(workflowReorderActionStepForm.getActionId(), user);
+
+            Logger.debug(this, "Looking for the stepId: "
+                    + workflowReorderActionStepForm.getStepId());
+            step =
+                    this.workflowAPI.findStep(workflowReorderActionStepForm.getStepId());
+
+            if (null == action) {
+                throw new DoesNotExistException("Workflow-does-not-exists-action");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Workflow-does-not-exists-action" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)